controllers/snapshot/snapshot_adapter.go

```diff
@@ -306,6 +307,27 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 					"integrationTestScenario.Name", integrationTestScenario.Name)
 				return controller.StopProcessing()
 			}
+			if clienterrors.IsInvalid(err) {
+
+				//a.logger.Error(err, "environment or deploymentTargetClass referenced by integrationTestScenario is invalid", "integrationTestScenario.Name", integrationTestScenario.Name)
+				if !gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
+					var (
+						errKind   string = "<unknown>"
+						errName   string = "<unknown>"
+						statusErr *clienterrors.StatusError
+					)
+					if ok := errors.As(err, &statusErr); ok {
+						errKind = statusErr.Status().Details.Kind
+						errName = statusErr.Status().Details.Name
+					}
+					a.logger.Error(err, fmt.Sprintf("Resource %s (%s) referenced by integrationTestScenario is invalid", errName, errKind), "integrationTestScenario.Name", integrationTestScenario.Name)
+					_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, fmt.Sprintf("Object %s (%s) associated with snapshot %s is invalid: %s.", errName, errKind, a.snapshot.Spec.Application, err))
+					if err != nil {
+						a.logger.Error(err, "Failed to Update Snapshot status")
+					}
```

Comment:
Do we want to call `controller.RequeueWithError(err)` here? or else, if there were an error while marking the snapshot as failed, the controller wouldn't retry and stop reconciling the snapshot. In the end, we could have snapshots in the cluster that aren't marked as failed yet, it could be misleading for the users.

Reply:
Ack yes you are correct, thanks for the good catch @dheerajodha. Updated in [b674ec](https://github.com/redhat-appstudio/integration-service/pull/421/commits/b674ecd242f54c9700dd14f99c04946a91f6af72)

Reply:
Thanks!